ext/pdo_pgsql/pgsql_driver.c

```diff
@@ -1224,6 +1242,34 @@ PHP_METHOD(PDO_PGSql_Ext, pgsqlGetPid)
 }
 /* }}} */
 
+/* {{{ proto bool PDO::pgsqlSetNoticeCallback(mixed callback)
+   Sets a callback to receive DB notices (after client_min_messages has been set) */
+PHP_METHOD(PDO_PGSql_Ext, pgsqlSetNoticeCallback)
+{
+	pdo_dbh_t *dbh;
+	pdo_pgsql_db_handle *H;
+	zend_fcall_info fci = empty_fcall_info;
+	zend_fcall_info_cache fcc = empty_fcall_info_cache;
+	
+	dbh = Z_PDO_DBH_P(getThis());
+	PDO_CONSTRUCT_CHECK;
+	
+	H = (pdo_pgsql_db_handle *)dbh->driver_data;
+	
+	pdo_pgsql_cleanup_notice_callback(H);
+	if (FAILURE == zend_parse_parameters(ZEND_NUM_ARGS(), "F!", &fci, &fcc)) {
+		RETURN_FALSE;
+	}
+	
+	if (ZEND_FCC_INITIALIZED(fcc)) {
+		H->notice_callback = ecalloc(1, sizeof(zend_fcall_info_cache));
```

Comment:
You don't have to use ecalloc here, emalloc suffices:

```suggestion
		H->notice_callback = emalloc(sizeof(zend_fcall_info_cache));
```